### DISCUSSION TITLE
Scaling celeryworker service each with different queue

### DISCUSSION THREAD

**Discussion by koztay** [cookiecutter/cookiecutter-django#3427] [Q&A] [answered]
I am using cookiecutter-django with docker and I want to scale my celeryworkers each with different queue. But could not find a way. If I use "--scale celeryworker=4" command it creates 4 different workers with same settings. But I need to set each one different queue. Is there anybody have the similar issue or have any idea about how to solve this problem? Any help would be very appreciated.
Thanks,

**Comment by vladdoster** [comment, marked as answer]
Hello @koztay ,

You might try looking at these articles

[article 1](https://hackernoon.com/using-celery-with-multiple-queues-retries-and-scheduled-tasks-589fe9a4f9ba) & [article 2](http://www.madhur.co.in/blog/2015/11/17/celery-tips-and-tricks.html)

Basically you want to create Celery routes which designates a task to a certain queue

`CELERY_ROUTES = {
    'core.tasks.too_long_task': {'queue': 'too_long_queue'},
    'core.tasks.quick_task': {'queue': 'quick_queue'},}`

Next, you can scale up queues as needed

#### Scaling queues

`celery --app=proj_name worker -Q too_long_queue -c 2`

`celery --app=proj_name worker -Q quick_queue -c 2 `
